kratos/solving_strategies/schemes/residual_based_bdf_custom_scheme.h

```diff
@@ -366,26 +366,45 @@ class ResidualBasedBDFCustomScheme
      * @brief Updating first time derivative (velocity)
      * @param itNode the node interator
      */
-
     inline void UpdateFirstDerivative(NodesArrayType::iterator itNode) override
     {
         // DOUBLES
         std::size_t counter = 0;
         for ( const auto& i_var : mDoubleVariable) {
-            double& dotun0 = itNode->FastGetSolutionStepValue(mFirtsDoubleDerivatives[counter]);
-            dotun0 = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
-            for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
-                dotun0 += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
+            if (!itNode->IsFixed(mFirtsDoubleDerivatives[counter])) {
+                double& dotun0 = itNode->FastGetSolutionStepValue(mFirtsDoubleDerivatives[counter]);
+                dotun0 = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
+                for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
+                    dotun0 += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
+            }
             counter++;
         }
 
         // ARRAYS
         counter = 0;
         for ( const auto& i_var : mArrayVariable) {
-            array_1d<double, 3>& dotun0 = itNode->FastGetSolutionStepValue(mFirtsArrayDerivatives[counter]);
-            noalias(dotun0) = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
+
+            const std::string& dvariable_name = (mFirtsArrayDerivatives[counter]).Name();
+            const VariableComponent<ComponentType>& dvar_x = KratosComponents< VariableComponent<ComponentType>>::Get(dvariable_name + "_X");
+            const VariableComponent<ComponentType>& dvar_y = KratosComponents< VariableComponent<ComponentType>>::Get(dvariable_name + "_Y");
+            const VariableComponent<ComponentType>& dvar_z = KratosComponents< VariableComponent<ComponentType>>::Get(dvariable_name + "_Z");
+
+            array_1d<double, 3> auxiliar_dotun0 =  BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
             for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
-                noalias(dotun0) += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
+                noalias(auxiliar_dotun0) += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
```

Comment:
You are computing this for all the nodes, including those that have true fixity in all the directions. Maybe it's better to retrieve the `i_var` value in here (just one memory access) and then to do the value computation inside the fixity if statements below. Am I wrong?

Reply:
It is an auxiliar array, the check is done later

Reply:
I now that the check is done later. What I mean is that for fixed DOFs you don't need to compute such auxiliary array. I was wondering if it is more efficient to:
1) Access the memory (to avoid multiple accesses in case there is no fixity)
2) Check fixity
3) From the previous accessed values, compute such auxiliary value

Reply:
Then I will need to check first the three components (I don't think I save that much)

Reply:
Actually not. But in any case I agree that you don't save that much.